crates/bevy_reflect/bevy_reflect_derive/src/impls/structs.rs

```diff
@@ -168,50 +175,38 @@ pub(crate) fn impl_struct(reflect_struct: &ReflectStruct) -> TokenStream {
                 <Self as #bevy_reflect_path::Typed>::type_info()
             }
 
-            #[inline]
-            fn into_any(self: #FQBox<Self>) -> #FQBox<dyn #FQAny> {
-                self
+            fn as_full(&self) -> #FQOption<&dyn #bevy_reflect_path::Reflect> {
+                Some(self)
```

Comment:
Technically this should be `#FQOption::Some` (or `#FQAny::Ok` in the case of `into_full`).

Same for all other methods and impl files.